### ISSUE TITLE
calling os.Exit(0) from a `gopherjs run` invocation fails with "fatal error: all goroutines are asleep - deadlock!"

### ISSUE THREAD

**Issue by nullstyle** [gopherjs/gopherjs#531]
Here's the repro script:

``` golang
package main

import (
    "fmt"
    "os"
)

func main() {
    fmt.Println("hello world")
    os.Exit(0)
}
```

sample results:

``` shell
➜  testj git:(mithril) ✗ gopherjs run ./main.go
gopherjs: Source maps disabled. Use Node.js 4.x with source-map-support module for nice stack traces.
hello world
fatal error: all goroutines are asleep - deadlock!
```

This appears to be the root cause of #395 as well.

**Comment by shurcooL** [comment]
Just checking, are you familiar with https://github.com/gopherjs/gopherjs#application-lifecycle?

**Comment by nullstyle** [comment]
My understanding was that only applies to the browser... `gopherjs test` runs using node, AFAIK.

**Comment by nullstyle** [comment]
You can see the workaround I'm using presently at https://github.com/nullstyle/gopherjs/commit/8d4d6d07a8d9acca07efebeba727363324c33004

**Comment by shurcooL** [comment]
I cannot reproduce with latest version of GopherJS.

I have Node version v6.7.0 with both node-syscall and source-map-support modules installed.

Running the program above gives me:

```bash
$ node --version
v6.7.0

$ go run main.go
hello world
$ gopherjs run main.go 
hello world
```

Can you try again with both node-syscall and source-map-support modules installed? I suspect that they might make a difference.

If so, this is definitely related to https://github.com/gopherjs/gopherjs/pull/449 (see discussion about source-map-support there).

**Comment by nullstyle** [comment]
after installing node-syscall, the issue no longer occurs.   I didn't even know node-syscall was a thing, seems like it should be in the README.  Are you open to me making a PR to add that?

Is there anyway to get node-syscall into npm?

**Comment by shurcooL** [comment]
Okay, thanks for confirming.

> I didn't even know node-syscall was a thing, seems like it should be in the README.

It is in the README, in the Installation and Usage section.

https://github.com/gopherjs/gopherjs#installation-and-usage

**Comment by nullstyle** [comment]
> It is in the README, in the Installation and Usage section.

IMO It's not clear that it's required to get basic behavior like `gopher test` working.

**Comment by hajimehoshi** [comment]
I'm not familiar with node, and I don't know how to install node-syscall...

**Comment by hajimehoshi** [comment]
Ah, I found https://github.com/gopherjs/gopherjs/blob/master/doc/syscalls.md

**Comment by shurcooL** [comment]
As far as I can tell, the only remaining actionable issue here was this:

> IMO It's not clear that it's required to get basic behavior like `gopherjs test` working.

I believe that has been fixed in 33af5e0039fd599c4b6b3c8a70e503a3b2f80a7e. Closing.
